### ISSUE TITLE
sanitize actix-web URI's

### ISSUE THREAD

**Issue by tiagolobocastro** [wafflespeanut/paperclip#293]
I'm not that knowledgable in openapi so I might be saying something wrong, but from what I've tested using an openapi generator it seems openapi does not like to see "tail match" in a URI, eg:

`foo/{bar}/{tail:.*}`

which is used in actix-web.
So, should we be sanitizing the URI's? Eg, would we turn `foo/{bar}/{tail:.*}` into `foo/{bar}/{tail}` ?

**Comment by tiagolobocastro** [comment]
@dunnock what do you think?

**Comment by dunnock** [comment]
I do not think we should change user's definition, but we can definitely emit warning or error if that contradicts specification. Is there article in open api spec where it is contradicting?

**Comment by tiagolobocastro** [comment]
It seems quite sparse in that regard: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#pathsObject
It mentions path templating (to use `{var_name}`) but doesn't go into any more details.

**Comment by tiagolobocastro** [comment]
hmm I installed the generator again, and this time it seems to have worked without me removing the `:.*`; maybe this was just a bug on the openapi-generator then.

**Comment by tiagolobocastro** [comment]
Just used a go client generated by the openapi-generator and it worked fine, so I'm closing this.
